lib/xhr.js

```diff
@@ -189,157 +179,169 @@ function enableXHROverH2(XMLHttpRequest, configuration) {
-                        port: destination.port,
-                        method: xhrInfo.get(self, 'method'),
-                        path: destination.path,
-                        headers: xhrInfo.get(self, 'headers').toObject(),
-                        // auth: self.__headers // TODO AUTH
-                        // TODO, change transport to createConnection
-                        transport: transport,
-                        transportUrl: proxyTransportUrl
-                        // TODO timeout if syncronization set
-                        // timeout: self.__timeout
-                    }, function (response) {
-
-                        // Export live response
-                        defineXhrResponse(self, response);
-
-                        self._changeState(XMLHttpRequest.HEADERS_RECEIVED, {'response': response});
-                        response.on('data', function (data) {
-                            response.data = mergeTypedArrays(response.data, data);
-                            self._changeState(XMLHttpRequest.LOADING, {'response': response});
-                        });
-                        response.on('finish', function () {
-                            if (configuration.debug) {
-                                configuration._log.debug("Got response for proxied XHR(" + destination.href + ")");
-                            }
-                            configuration.cache.put(requestInfo, response).then(function () {
-                                configuration._log.debug("Cache updated for proxied XHR(" + destination.href + ")");
-                            }, function (cacheError) {
-                                configuration._log.debug("Cache error for proxied XHR(" + destination.href + "):" + cacheError.message);                            
-                            }).finally(function () {
-                                self._changeState(XMLHttpRequest.DONE, {
-                                    'response': response
-                                });
+                        configuration.cache.put(requestInfo, response).then(function () {
+                            configuration._log.debug("Cache updated for proxied XHR(" + destination.href + ")");
+                        }, function (cacheError) {
+                            configuration._log.debug("Cache error for proxied XHR(" + destination.href + "):" + cacheError.message);                            
+                        }).finally(function () {
+                            self._changeState(XMLHttpRequest.DONE, {
+                                'response': response
                             });
                         });
                     });
+                });
 
-                    request.on('error', function (/*e*/) {
-                        // TODO, handle error
-                        // self._changeState('error');
-                    });
+                // TODO Why ?
```

Comment:
I added the original todo thinking it should result in `https://developer.mozilla.org/en-US/docs/Web/Events/error`